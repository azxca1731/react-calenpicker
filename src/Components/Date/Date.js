import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import startImg from "Styles/assets/start-period.png";
import endImg from "Styles/assets/end-period.png";

const DateTd = styled.td`
  display: inline-block;
  padding-left: 0.3%;
  padding-right: 0.3%;
  width: 13.6%;
  height: 100%;
  text-align: center;
  position: relative;
  font-weight: bold;
  background-color: ${props => (props.isInPeriod ? props.theme.secondaryColor : props.theme.backgroundColor)};
`;

const DateDiv = styled.div`
  color: ${props => {
    const { isHoliday, isToday, dayNumber, isSaturday } = props;
    if (dayNumber == 0 || isHoliday) return props.theme.holidayColor;
    else if (isToday) return props.theme.todayColor;
    else if (isSaturday) return props.theme.saturdayColor;
    else return props.theme.fontColor;
  }};
  opacity: ${props => (props.isInThisMonth ? 1 : 0.3)};
`;

const DateTextDiv = styled.div`
  width: inherit;
  height: 100%;
  overflow-x: hidden;
  font-size: 80%;
`;

const DateDayNumberDiv = styled.div`
  margin-bottom: 20%;
`;

const EndPointDiv = styled.div`
  height: 100%;
  background: url(${props => props.img}) no-repeat;
  background-size: 100% 100%;
`;

const StartIndicator = props => (
  <EndPointDiv img={startImg}>
    <DateDayNumberDiv>{props.dayNumber}</DateDayNumberDiv>
    <DateTextDiv>시작</DateTextDiv>
  </EndPointDiv>
);

const EndIndicator = props => (
  <EndPointDiv img={endImg}>
    <DateDayNumberDiv>{props.dayNumber}</DateDayNumberDiv>
    <DateTextDiv>끝</DateTextDiv>
  </EndPointDiv>
);

const DateIndicator = props => {
  const handleTextClicked = event => {
    const { dateString, text, isHoliday } = props;
    props.handleModal("READ");
    props.handleTargetSetValue({ date: dateString, text, isHoliday });
    event.preventDefault();
    event.stopPropagation();
  };

  return (
    <DateDiv isHoliday={props.isHoliday} isToday={props.isToday} isInThisMonth={props.isInThisMonth} isSaturday={props.isSaturday} dayNumber={props.dayNumber}>
      <DateDayNumberDiv>{props.dayNumber}</DateDayNumberDiv>
      <DateTextDiv onClick={handleTextClicked}>{props.text}</DateTextDiv>
      {props.haveMoreDate ? <DateTextDiv>...</DateTextDiv> : null}
    </DateDiv>
  );
};

const SelectIndicator = props => (
  <DateDiv isHoliday={props.isHoliday} isToday={props.isToday} isInThisMonth={props.isInThisMonth} isSaturday={props.isSaturday} dayNumber={props.dayNumber}>
    <DateDayNumberDiv>{props.dayNumber}</DateDayNumberDiv>
    <DateTextDiv>{props.text ? props.text : "선택"}</DateTextDiv>
  </DateDiv>
);

const Date = props => {
  const { isInPeriod, dayNumber, isHoliday, isInThisMonth, isToday, isSaturday, cssObject, text, indicatorType, handleDateClick, handleModal, dateString, handleTargetSetValue, haveMoreDate } = props;
  let contents;
  if (indicatorType == "date") {
    contents = DateIndicator({ isHoliday, isToday, isInThisMonth, isSaturday, dayNumber, text, handleModal, dateString, handleTargetSetValue, haveMoreDate });
  } else if (indicatorType == "start") {
    contents = StartIndicator({ dayNumber });
  } else if (indicatorType == "end") {
    contents = EndIndicator({ dayNumber });
  } else {
    contents = SelectIndicator({ isHoliday, isToday, isInThisMonth, isSaturday, dayNumber, text });
  }
  return (
    <DateTd onClick={handleDateClick} style={cssObject} isInPeriod={isInPeriod}>
      {contents}
    </DateTd>
  );
};
Date.defaultProps = {
  handlers: {
    handleDateClick: () => {}
  },
  cssObject: {},
  dayNumber: 1,
  indicatorType: "date",
  isInPeriod: false,
  isInThisMonth: false
};

StartIndicator.propTypes = {
  dayNumber: PropTypes.number
};

EndIndicator.propTypes = {
  dayNumber: PropTypes.number
};

DateIndicator.propTypes = {
  isHoliday: PropTypes.bool,
  isToday: PropTypes.bool,
  isInThisMonth: PropTypes.bool,
  isSaturday: PropTypes.bool,
  dayNumber: PropTypes.number,
  text: PropTypes.string,
  handleModal: PropTypes.func,
  dateString: PropTypes.string,
  handleTargetSetValue: PropTypes.func,
  haveMoreDate: PropTypes.haveMoreDate
};

SelectIndicator.propTypes = {
  isHoliday: PropTypes.bool,
  isToday: PropTypes.bool,
  isInThisMonth: PropTypes.bool,
  isSaturday: PropTypes.bool,
  dayNumber: PropTypes.number,
  text: PropTypes.string
};

Date.propTypes = {
  handleDateClick: PropTypes.func,
  cssObject: PropTypes.object,
  dayNumber: PropTypes.number,
  text: PropTypes.string,
  style: PropTypes.any,
  isInPeriod: PropTypes.bool,
  isInThisMonth: PropTypes.bool,
  isHoliday: PropTypes.bool,
  isToday: PropTypes.bool,
  isSaturday: PropTypes.bool,
  indicatorType: PropTypes.oneOf(["date", "start", "end", "select"]),
  handleModal: PropTypes.func,
  dateString: PropTypes.string,
  handleTargetSetValue: PropTypes.func,
  haveMoreDate: PropTypes.bool
};

export default Date;
