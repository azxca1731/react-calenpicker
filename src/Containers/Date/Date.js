import React from "react";
import PropTypes from "prop-types";
import style from "./Date.style";
import { DayConnector, PropsConnector, CssConnector } from "../Provider";

class Date extends React.Component {
  constructor(props) {
    super(props);
    this.handleDateClick = this.handleDateClick.bind(this);
    this.setClassName = this.setClassName.bind(this);
    this.handleInPeriod = this.handleInPeriod.bind(this);
    this.handleStart = this.handleStart.bind(this);
    this.handleEnd = this.handleEnd.bind(this);
  }

  state = {
    today: this.props.getTodayString(),
    dateString: "",
    dayNumber: 0,
    isInThisMonth: false
  };

  static getDerivedStateFromProps(props, state) {
    const {
      weekNumber,
      day,
      objectSetText,
      duplicated,
      duplicatedDateObjectArray,
      indicateToday
    } = props;
    const dateObjectArray = duplicated
      ? duplicatedDateObjectArray
      : props.dateObjectArray;
    if (dateObjectArray.length > 0) {
      const target = dateObjectArray[weekNumber * 7 + day - 1];
      const dateString = target.dateString;
      const isInThisMonth = target.isInThisMonth;
      let text, isHoliday;
      if (state.today === dateString && indicateToday) {
        text = "오늘";
      } else {
        const filtered = objectSetText.filter(item => {
          return item.date === dateString;
        });
        text = filtered.length > 0 ? filtered[0].text : "";
        isHoliday = filtered.length > 0 ? filtered[0].isHoliday : false;
      }
      const dayNumber = target.dayNumber;
      return { dateString, text, dayNumber, isInThisMonth, isHoliday };
    }
    return null;
  }

  componentDidMount() {}

  setClassName() {
    let className;
    const { day, indicateToday, onlyThisMonth } = this.props;
    const { today, dateString, isInThisMonth, isHoliday } = this.state;
    if (day % 6 !== 1) {
      className = isHoliday ? style["Date--sun"] : style["Date--day"];
    } else if (day == 1) {
      className = style["Date--sun"];
    } else {
      className = style["Date--sat"];
    }
    if (!isInThisMonth) {
      className += ` ${style["Date--past"]}`;
    }
    if (indicateToday && dateString == today) {
      className += ` ${style["Date--today"]}`;
    }
    if (onlyThisMonth && !isInThisMonth) {
      className = style["Date--notThisMonth"];
    }
    return className;
  }

  handleDateClick() {
    const { dateClicked } = this.props;
    const { isInThisMonth } = this.state;
    if (isInThisMonth) {
      dateClicked(this.state);
    }
  }

  handleInPeriod() {
    let className = style.Date;
    const { isInPeriod } = this.props;
    const { dateString } = this.state;
    if (isInPeriod(dateString)) {
      className += ` ${style["Date--periodSelect"]}`;
    }
    return className;
  }

  handleStart() {
    const { periodStart, periods } = this.props;

    const { dateString, dayNumber } = this.state;

    if (
      (periodStart && dateString === periodStart) ||
      (periods.length > 0 &&
        periods.filter(({ periodStart }) => periodStart === dateString).length >
          0)
    ) {
      return <div className={style.Date__periodStart}>{dayNumber}</div>;
    }
  }

  handleEnd() {
    const { periodEnd, periods } = this.props;
    const { dateString, dayNumber } = this.state;
    if (
      (periodEnd && dateString === periodEnd) ||
      (periods.length > 0 &&
        periods.filter(({ periodEnd }) => periodEnd === dateString).length > 0)
    ) {
      return <div className={style.Date__periodEnd}>{dayNumber}</div>;
    }
  }

  render() {
    const { cssObject } = this.props;
    const { text, dayNumber } = this.state;
    return (
      <td
        className={this.handleInPeriod()}
        onClick={this.handleDateClick}
        style={cssObject}
      >
        <div className={this.setClassName()}>
          {dayNumber}
          <div className={style["Date--text"]}>{text}</div>
          {this.handleStart()}
          {this.handleEnd()}
        </div>
      </td>
    );
  }
}

Date.defaultProps = {
  day: "1"
};

Date.propTypes = {
  weekNumber: PropTypes.number.isRequired,
  day: PropTypes.oneOf([1, 2, 3, 4, 5, 6, 7]).isRequired,
  dateObjectArray: PropTypes.array.isRequired,
  dateClicked: PropTypes.func,
  onlyThisMonth: PropTypes.bool,
  isInPeriod: PropTypes.func,
  objectSetText: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
      date: PropTypes.date
    })
  ),
  periodStart: PropTypes.string,
  periodEnd: PropTypes.string,
  getTodayString: PropTypes.func,
  indicateToday: PropTypes.bool,
  multiSelect: PropTypes.bool,
  periods: PropTypes.arrayOf(
    PropTypes.shape({
      periodStart: PropTypes.string,
      periodEnd: PropTypes.string
    })
  ),
  duplicated: PropTypes.bool,
  duplicatedDateObjectArray: PropTypes.array,
  cssObject: PropTypes.object
};

export default PropsConnector(({ state }) => ({
  onlyThisMonth: state.onlyThisMonth,
  objectSetText: state.objectSetText,
  duplicated: state.duplicated
}))(
  DayConnector(({ state, actions }) => ({
    dateObjectArray: state.dateObjectArray,
    duplicatedDateObjectArray: state.duplicatedDateObjectArray,
    dateClicked: actions.handleDateClicked,
    isInPeriod: actions.isInPeriod,
    periodStart: state.periodStart,
    periodEnd: state.periodEnd,
    getTodayString: actions.getTodayString,
    indicateToday: state.indicateToday,
    multiSelect: state.multiSelect,
    periods: state.periods
  }))(
    CssConnector(({ state }) => ({
      cssObject: state.DateCssObject
    }))(Date)
  )
);
