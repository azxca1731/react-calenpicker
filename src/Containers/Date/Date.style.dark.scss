@import "Styles/theme/_dark.scss";
@import "./Date.style.scss";