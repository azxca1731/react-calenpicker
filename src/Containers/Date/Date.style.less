@import "../../Styles/colors.less";
.Date--dark {
  display: inline-block;
  padding-left: 0.3%;
  padding-right: 0.3%;
  width: 13.6%;
  height: 100%;
  text-align: center;
  position: relative;
  &--day {
    color: white;
  }
  &--sun {
    color: red;
  }
  &--sat {
    color: blue;
  }
  &--past {
    opacity: 0.3;
  }
  &--periodSelect {
    background-color: #2d3436;
  }
  &--today {
    color: green;
  }
  &--text {
    width: inherit;
    height: 100%;
    justify-content: center;
    overflow-x: hidden;
  }
  &__periodStart {
    width: 100%;
    height: 100%;
    position: absolute;
    left: 0;
    top: 0%;
    background-color: #b2bec3;
    border-top-left-radius: 200px 200px;
    border-bottom-left-radius: 200px 200px;
    color: white;
  }
  &__periodEnd {
    width: 100%;
    height: 100%;
    position: absolute;
    left: 0;
    top: 0%;
    background-color: #b2bec3;
    border-top-right-radius: 200px 200px;
    border-bottom-right-radius: 200px 200px;
    color: white;
  }
}

.Date {
  display: inline-block;
  padding-left: 0.3%;
  padding-right: 0.3%;
  width: 13.6%;
  height: 100%;
  text-align: center;
  position: relative;
  &--day {
    color: @font-color;
  }
  &--sun {
    color: @holiday-color;
  }
  &--sat {
    color: @saturday-color;
  }
  &--past {
    opacity: 0.3;
  }
  &--notThisMonth {
    display: none;
  }
  &--periodSelect {
    background-color: @secondary-color;
  }
  &--today {
    color: @today-color;
  }
  &--text {
    width: inherit;
    height: 100%;
    justify-content: center;
    overflow-x: hidden;
  }
  &__periodStart {
    width: 100%;
    height: 100%;
    position: absolute;
    left: 0;
    top: 0%;
    background-color: @primary-color;
    border-top-left-radius: 200px 200px;
    border-bottom-left-radius: 200px 200px;
    color: @selected-font-color;
  }
  &__periodEnd {
    width: 100%;
    height: 100%;
    position: absolute;
    left: 0;
    top: 0%;
    background-color: @primary-color;
    border-top-right-radius: 200px 200px;
    border-bottom-right-radius: 200px 200px;
    color: @selected-font-color;
  }
}