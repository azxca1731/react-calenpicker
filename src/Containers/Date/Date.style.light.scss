@import "../../Styles/theme/_light.scss";
@import "./Date.style.scss";
