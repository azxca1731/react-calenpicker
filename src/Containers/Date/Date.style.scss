.Date {
  display: inline-block;
  padding-left: 0.3%;
  padding-right: 0.3%;
  width: 13.6%;
  height: 100%;
  text-align: center;
  position: relative;

  &--day {
    color: black;
  }

  &--sun {
    color: red;
  }

  &--sat {
    color: blue;
  }

  &--past {
    opacity: 0.3;
  }

  &--periodSelect {
    background-color: beige;
  }

  &--today {
    color: green;
  }

  &--text {
    width: inherit;
    height: 100%;
    justify-content: center;
    overflow-x: hidden;
  }

  &__periodStart {
    width: 100%;
    height: 100%;
    position: absolute;
    left: 0;
    top: 0%;
    background-color: bisque;
    color: white;
  }

  &__periodEnd {
    width: 100%;
    height: 100%;
    position: absolute;
    left: 0;
    top: 0%;
    background-color: bisque;
    color: white;
  }
}
