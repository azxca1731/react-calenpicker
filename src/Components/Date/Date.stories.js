import React from "react";
import { ThemeProvider } from "styled-components";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";

import light from "Styles/theme/light";
import dark from "Styles/theme/dark";
import Date from "Components/Date";

storiesOf("Components/Date", module)
  .addDecorator(withInfo)
  .addDecorator(story => <ThemeProvider theme={light}>{story()}</ThemeProvider>)
  .add("Light", () => <Date />);

storiesOf("Components/Date", module)
  .addDecorator(withInfo)
  .addDecorator(story => <ThemeProvider theme={dark}>{story()}</ThemeProvider>)
  .add("Dark", () => <Date />);
