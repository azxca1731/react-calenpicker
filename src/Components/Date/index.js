import Date, { DateTextDIV } from "./Date";
export const DateText = DateTextDIV;
export default Date;
