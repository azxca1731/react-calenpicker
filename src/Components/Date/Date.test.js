import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import Date from "./index";

describe("Month", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<Date />);
    expect(wrapper).toHaveLength(1);
  });

  it("renders with snapshot", () => {
    const wrapper = shallow(<Date />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("render custom css render well with snapshot", () => {
    const customCssObject = { color: "red" };
    const wrapper = shallow(<Date cssObject={customCssObject} />);
    expect(wrapper).toMatchSnapshot();
  });
});
