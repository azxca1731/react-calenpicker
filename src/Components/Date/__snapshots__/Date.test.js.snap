// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Month render custom css render well with snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Date
    cssObject={
      Object {
        "color": "red",
      }
    }
    dayNumber={1}
    handlers={
      Object {
        "handleDateClick": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef
        dayNumber={1}
      >
        <ForwardRef>
          1
        </ForwardRef>
        <ForwardRef>
          선택
        </ForwardRef>
      </ForwardRef>,
      "isInPeriod": undefined,
      "onClick": undefined,
      "style": Object {
        "color": "red",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            1
          </ForwardRef>,
          <ForwardRef>
            선택
          </ForwardRef>,
        ],
        "dayNumber": 1,
        "isHoliday": undefined,
        "isInThisMonth": undefined,
        "isSaturday": undefined,
        "isToday": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": 1,
          },
          "ref": null,
          "rendered": 1,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": true,
              "rules": Array [
                "
  margin-bottom: 20%;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "선택",
          },
          "ref": null,
          "rendered": "선택",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": true,
              "rules": Array [
                "
  width: inherit;
  height: 100%;
  justify-content: center;
  overflow-x: hidden;
  font-size: 80%;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "rules": Array [
            "
  color: ",
            [Function],
            ";
  opacity: ",
            [Function],
            ";
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  display: inline-block;
  padding-left: 0.3%;
  padding-right: 0.3%;
  width: 13.6%;
  height: 100%;
  text-align: center;
  position: relative;
  font-weight: bold;
  background-color: ",
          [Function],
          ";
",
        ],
      },
      "displayName": "styled.td",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "td",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef
          dayNumber={1}
        >
          <ForwardRef>
            1
          </ForwardRef>
          <ForwardRef>
            선택
          </ForwardRef>
        </ForwardRef>,
        "isInPeriod": undefined,
        "onClick": undefined,
        "style": Object {
          "color": "red",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              1
            </ForwardRef>,
            <ForwardRef>
              선택
            </ForwardRef>,
          ],
          "dayNumber": 1,
          "isHoliday": undefined,
          "isInThisMonth": undefined,
          "isSaturday": undefined,
          "isToday": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": 1,
            },
            "ref": null,
            "rendered": 1,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bxivhb",
                "isStatic": true,
                "rules": Array [
                  "
  margin-bottom: 20%;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bxivhb",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "선택",
            },
            "ref": null,
            "rendered": "선택",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-htpNat",
                "isStatic": true,
                "rules": Array [
                  "
  width: inherit;
  height: 100%;
  justify-content: center;
  overflow-x: hidden;
  font-size: 80%;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-htpNat",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
  color: ",
              [Function],
              ";
  opacity: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  display: inline-block;
  padding-left: 0.3%;
  padding-right: 0.3%;
  width: 13.6%;
  height: 100%;
  text-align: center;
  position: relative;
  font-weight: bold;
  background-color: ",
            [Function],
            ";
",
          ],
        },
        "displayName": "styled.td",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "td",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Month renders with snapshot 1`] = `
<styled.td
  style={Object {}}
>
  <styled.div
    dayNumber={1}
  >
    <styled.div>
      1
    </styled.div>
    <styled.div>
      선택
    </styled.div>
  </styled.div>
</styled.td>
`;
